lib/matplotlib/backend_managers.py

```diff
@@ -56,14 +56,13 @@ class ToolManager(object):
         `LockDraw` object to know if the message is available to write
     """
 
-    def __init__(self, canvas):
+    def __init__(self, manager):
```

Comment:
Nothing stops you from creating a tool that interacts with the FigureManager instead of a Figure and I like that idea wide open. 
Think about adding/removing one toolbar from another, this has to go throught the FigureManager
Or for example once we get rid of Gcf `ToolQuit` and `ToolQuitAll` will have to talk to manager instead of Gcf